src/App.tsx

```diff
@@ -1,13 +1,29 @@
 import React from 'react';
+import {
+  Switch,
+  Route,
+} from "react-router-dom";
 import Home from './components/Home';
-
+import Settings from './components/Settings';
+import Header from './components/Header';
+import Footer from './components/Footer';
+import Login from './components/Login';
+import Article from './components/Article';
 
 
 function App() {
   return (
+    <main><Header />
```

Comment:
If you are needing to change the `<Header>` and <`Footer>` based on the route you would need to put those components in the page level components. You will not be able to use the router based hooks if these components are outside the router.

Reply:
I would like to discuss this more. The current state of the branch - with header and footer in each page - allows conditional formatting of links based on route. Seems like they should belong in the App component though, which is why I wanted to discuss further.

Reply:
I came back to this and resolved it. moved header and footer back to App inside router and outside switch.